<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2025-05-15" />

<title>CDsubtypes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2025_adultchilddementiagenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="adultvCD.html">Adult_all_and_CD_all</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">CDsubtypes</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2025-05-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>2025_adultchilddementiagenetics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250515code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250515code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250515)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong4d75fce">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 4d75fce
</a>
</p>
</div>
<div id="strongRepositoryversionstrong4d75fce"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 4d75fce.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  VennDiagram.2025-05-15_11-46-54.361617.log
    Untracked:  VennDiagram.2025-05-15_11-47-06.913106.log
    Untracked:  VennDiagram.2025-05-15_11-47-13.625688.log
    Untracked:  VennDiagram.2025-05-15_11-47-26.446071.log
    Untracked:  analysis/CDsubtypes.rmd
    Untracked:  analysis/adultvCD.rmd
    Untracked:  data/R_objects/
    Untracked:  data/genesets/
    Untracked:  data/gtex/
    Untracked:  data/omicsynth/
    Untracked:  data/results/

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>knitr::opts_chunk$set(
  message = F, echo = T, warning = F
)</code></pre>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(readxl)

# vis
library(ComplexUpset)
library(UpSetR)
library(pander)
library(ggeasy)
library(ggpubr)
library(ggrepel)
library(plotly)
library(tidyHeatmap)
library(grid)
library(gridExtra)
library(gridGraphics)
library(viridis)
library(tidytext)

# annotations
library(org.Hs.eg.db)
library(ndexr)

# enrichemnt
library(clusterProfiler)
library(ReactomePA)

theme_set(theme_light() +
            theme(plot.title = element_text(hjust = 0.5), 
                  strip.text = element_text(face = &quot;bold&quot;),
                  strip.background = element_rect(fill = &quot;grey20&quot;)
            )
)

knitr::opts_chunk$set(
  message = F, echo = F, warning = F
)</code></pre>
</div>
<div id="import-gene-sets" class="section level1">
<h1>import gene sets</h1>
<p>tony curated these genesets from clingen. here is the number of genes
per gene set.</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="overlap-of-genes-across-gene-sets" class="section level2">
<h2>overlap of genes across gene sets</h2>
<p>There is minimal overlap of genes across the adult and childhod
dementia gene sets. Just 1 or 2 in each comparison</p>
<p>the 10 gene overlap is between ALS and FTLD</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="over-representation-analysis-of-gene-sets"
class="section level1">
<h1>over-representation analysis of gene sets</h1>
<p>Here I will perform pathway/enrichment analysis on these gene sets to
see what pathways are enriched, and how much overlap there is.</p>
<p>the background gene set will be all genes found in the PCNet, as
these are the ones I am analysing. The node info was obtained from
NDex.</p>
<div id="go-terms" class="section level2">
<h2>GO terms</h2>
<p>Here, i performed over-represetation analysis (hypergeometric test)
of GO terms (all ontologies) of each gene set using clusterprofiler</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="kegg-pathways" class="section level2">
<h2>KEGG pathways</h2>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="reactome-pathways" class="section level2">
<h2>reactome pathways</h2>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="is-there-significant-colocalisation" class="section level1">
<h1>Is there significant colocalisation?</h1>
<p>Netcoloc was performed and below shows the expected vs observed ratio
of the genes identified by dual network propogation.</p>
<div id="plot-observed-v-expected-per-comparison"
class="section level2">
<h2>plot observed v expected per comparison</h2>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-observed-v-expected-ratio" class="section level2">
<h2>plot observed v expected ratio</h2>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="are-the-subnetworks-biologically-meaningful"
class="section level1">
<h1>Are the subnetworks biologically meaningful?</h1>
<p>Here, I will only consider the subnetworks which are significant</p>
<div
id="check-the-genes-in-the-subnetwork-are-enriched-with-mouse-brain-phenotypes"
class="section level2">
<h2>check the genes in the subnetwork are enriched with mouse brain
phenotypes</h2>
<p>I tested the enrichment of genes associated with mouse brain/neural
phenotypes in the NetColoc subnetwork as a <em>whole</em>. I also
included here a negative control phenotype - abnormal digit morphology.
This is not expected to be associated with the subnetworks.</p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-16-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="are-the-genes-in-the-subnetwork-human-brain-expressed"
class="section level2">
<h2>are the genes in the subnetwork human brain expressed?</h2>
<p>I next want to see whether the genes in the subnetwork are expressed
in the brain. To do this, I will assess it in the GTEx dataset. I will
use the GTEx v10 data, which is the most recent version.</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="obtain-gtex-expression-data" class="section level3">
<h3>obtain GTEx expression data</h3>
<p>This file was downloaded from here : <a
href="https://www.gtexportal.org/home/downloads/adult-gtex/bulk_tissue_expression"
class="uri">https://www.gtexportal.org/home/downloads/adult-gtex/bulk_tissue_expression</a></p>
<p>The file contains the median gene-level TPM by tissue.</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="what-pathways-are-in-the-subnetwork" class="section level1">
<h1>what pathways are in the subnetwork</h1>
<p>I will here show the top pathways identified in the clusters within
the subnetworks.</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-22-12.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="what-genes-are-in-the-subnetworks-which-are-drug-targetable"
class="section level1">
<h1>what genes are in the subnetworks which are drug targetable?</h1>
<p>Here, i obtained the drug targets data from <a
href="https://doi.org/10.1016/j.ajhg.2023.12.006">omicsynth</a> to see
what genes in the subnetwork have a known drug target.</p>
<p>The plots below show the network proximity score (higher means more
proximal to network) vs the evidence of drug interaction (higher
interaction = higher score)</p>
<p><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/CDsubtypes.rmd/unnamed-chunk-23-12.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ReactomePA_1.50.0      clusterProfiler_4.14.6 ndexr_1.28.0          
 [4] RCX_1.10.0             org.Hs.eg.db_3.20.0    AnnotationDbi_1.68.0  
 [7] IRanges_2.40.1         S4Vectors_0.44.0       Biobase_2.66.0        
[10] BiocGenerics_0.52.0    tidytext_0.4.2         viridis_0.6.5         
[13] viridisLite_0.4.2      gridGraphics_0.5-1     gridExtra_2.3         
[16] tidyHeatmap_1.11.6     plotly_4.10.4          ggrepel_0.9.6         
[19] ggpubr_0.6.0           ggeasy_0.1.5           pander_0.6.6          
[22] UpSetR_1.4.0           ComplexUpset_1.3.3     readxl_1.4.5          
[25] magrittr_2.0.3         lubridate_1.9.4        forcats_1.0.0         
[28] stringr_1.5.1          dplyr_1.1.4            purrr_1.0.4           
[31] readr_2.1.5            tidyr_1.3.1            tibble_3.2.1          
[34] ggplot2_3.5.1          tidyverse_2.0.0       

loaded via a namespace (and not attached):
  [1] splines_4.4.2           later_1.4.1             ggplotify_0.1.2        
  [4] R.oo_1.27.0             cellranger_1.1.0        polyclip_1.10-7        
  [7] graph_1.84.1            lifecycle_1.0.4         rstatix_0.7.2          
 [10] doParallel_1.0.17       rprojroot_2.0.4         vroom_1.6.5            
 [13] lattice_0.22-6          MASS_7.3-65             dendextend_1.19.0      
 [16] backports_1.5.0         SnowballC_0.7.1         sass_0.4.9             
 [19] rmarkdown_2.29          jquerylib_0.1.4         yaml_2.3.10            
 [22] httpuv_1.6.15           ggtangle_0.0.6          cowplot_1.1.3          
 [25] DBI_1.2.3               RColorBrewer_1.1-3      abind_1.4-8            
 [28] zlibbioc_1.52.0         R.utils_2.13.0          ggraph_2.2.1           
 [31] yulab.utils_0.2.0       rappdirs_0.3.3          tweenr_2.0.3           
 [34] git2r_0.36.2            circlize_0.4.16         GenomeInfoDbData_1.2.13
 [37] enrichplot_1.26.6       tokenizers_0.3.0        tidytree_0.4.6         
 [40] reactome.db_1.89.0      codetools_0.2-20        ggforce_0.4.2          
 [43] DOSE_4.0.0              tidyselect_1.2.1        shape_1.4.6.1          
 [46] aplot_0.2.5             UCSC.utils_1.2.0        farver_2.1.2           
 [49] matrixStats_1.5.0       jsonlite_2.0.0          GetoptLong_1.0.5       
 [52] tidygraph_1.3.1         Formula_1.2-5           iterators_1.0.14       
 [55] foreach_1.5.2           tools_4.4.2             treeio_1.30.0          
 [58] Rcpp_1.0.14             glue_1.8.0              xfun_0.51              
 [61] qvalue_2.38.0           GenomeInfoDb_1.42.3     withr_3.0.2            
 [64] fastmap_1.2.0           digest_0.6.37           timechange_0.3.0       
 [67] R6_2.6.1                colorspace_2.1-1        Cairo_1.6-2            
 [70] GO.db_3.20.0            RSQLite_2.3.9           R.methodsS3_1.8.2      
 [73] generics_0.1.3          data.table_1.17.0       graphlayouts_1.2.2     
 [76] httr_1.4.7              htmlwidgets_1.6.4       graphite_1.52.0        
 [79] pkgconfig_2.0.3         gtable_0.3.6            blob_1.2.4             
 [82] workflowr_1.7.1         ComplexHeatmap_2.22.0   XVector_0.46.0         
 [85] janeaustenr_1.0.0       htmltools_0.5.8.1       carData_3.0-5          
 [88] fgsea_1.32.4            clue_0.3-66             scales_1.3.0           
 [91] png_0.1-8               ggfun_0.1.8             knitr_1.50             
 [94] rstudioapi_0.17.1       tzdb_0.5.0              reshape2_1.4.4         
 [97] rjson_0.2.23            nlme_3.1-167            cachem_1.1.0           
[100] GlobalOptions_0.1.2     parallel_4.4.2          pillar_1.10.1          
[103] vctrs_0.6.5             promises_1.3.2          car_3.1-3              
[106] cluster_2.1.8.1         evaluate_1.0.3          cli_3.6.4              
[109] compiler_4.4.2          rlang_1.1.5             crayon_1.5.3           
[112] ggsignif_0.6.4          labeling_0.4.3          plyr_1.8.9             
[115] fs_1.6.5                stringi_1.8.7           BiocParallel_1.40.0    
[118] munsell_0.5.1           Biostrings_2.74.1       lazyeval_0.2.2         
[121] GOSemSim_2.32.0         Matrix_1.7-3            hms_1.1.3              
[124] patchwork_1.3.0         bit64_4.6.0-1           KEGGREST_1.46.0        
[127] igraph_2.1.4            broom_1.0.8             memoise_2.0.1          
[130] bslib_0.9.0             ggtree_3.14.0           fastmatch_1.1-6        
[133] bit_4.6.0               ape_5.8-1               gson_0.1.0             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
