---
title: "adult_all vs CD"
author: "Karissa Barthelson"
date: "2025-05-15"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction
```{r}
knitr::opts_chunk$set(
  message = F, echo = F, warning = F
)
```

```{r}
library(tidyverse)
library(magrittr)
library(readxl)

# vis
library(ComplexUpset)
library(UpSetR)
library(pander)
library(ggeasy)
library(ggpubr)
library(ggrepel)
library(plotly)
library(tidyHeatmap)
library(grid)
library(gridExtra)
library(gridGraphics)
library(viridis)
library(tidytext)

# annotations
library(org.Hs.eg.db)
library(ndexr)

# enrichemnt
library(clusterProfiler)
library(ReactomePA)

theme_set(theme_light() +
            theme(plot.title = element_text(hjust = 0.5), 
                  strip.text = element_text(face = "bold"),
                  strip.background = element_rect(fill = "grey20")
            )
)
```

Here, I will perform the netcoloc analysis just comparing adult and childhood onset dementia, using brain malformations as a control. 

# gene sets analysed
```{r}
genesets <- list.files(
  path = "data/genesets/adultchild",
  pattern = ".csv",
  full.names = TRUE
) %>% 
  sapply(function(x) {
    x %>% 
      read_csv() %>% 
      .$GENE
  }, simplify = F) %>% 
  setNames(names(.) %>% 
             str_remove(pattern = "data/genesets/adultchild/") %>% 
             str_remove(pattern = ".csv"))
```


```{r}
genesets %>% 
  unlist %>% 
  as.data.frame(col.names = "gene") %>% 
  rownames_to_column("geneset") %>%
  set_colnames(c("geneset", "gene")) %>% 
  mutate(
   geneset = str_remove(geneset, pattern = "\\d{1,3}$"), 
    geneset = fct_infreq(geneset)  # Reorder by frequency
   ) %>% 
  ggplot(
    aes(x = geneset)
  ) +
  geom_bar() +
  easy_rotate_x_labels(angle = -45) +
  labs(
    title = "Number of genes per gene set", 
    y = "# genes"
  )
```

Only 10 genes are shared between adult- and childhood-onset dementia
```{r}
genesets %>% 
  fromList %>% 
  UpSetR::upset()
```

# over-representation analysis of gene sets

Here I will perform pathway/enrichemtn analysis on these gene sets to see what pathways are enriched, and how much overlap there is. 

the background gene set will be all genes found in the PCNet, as these are the ones I am analysing. The node info was obtained from NDex. 
```{r}
# ~~ only run this on the first time ~~#
#
# login to the NDEx server - run this in console
# ndexcon <- ndex_connect(
#  "karissa.barthelson@flinders.edu.au", "password - capital and 21")

# search the networks for 'pcNet v1.3'
# pcnet1.3 = "4de852d9-9908-11e9-bcaf-0ac135e8bacf"

# obtain the nodes
# nodes = ndex_network_get_aspect(ndexcon, pcnet1.3, 'nodes') %>% .$name

# save this out for reimporting later
# saveRDS(nodes, "data/R_objects/network/nodes.rds")

nodes <- readRDS("data/R_objects/network/nodes.rds")
```

## GO terms

Here, i performed over-represetation analysis (hypergeometric test) of GO terms (all ontologies) of each gene set using clusterprofiler

```{r}
runGO = function(geneset) {
  enrichGO(
    gene = geneset, 
    universe = nodes, 
    OrgDb = org.Hs.eg.db, 
    keyType = "SYMBOL", 
    ont = "ALL"
  )
}
```

```{r}
GO <- genesets %>% 
  sapply(runGO, simplify = F) %>% 
  sapply(as_tibble, simplify = F)

GO %>% 
  lapply(function(x) {
    x$Description
  }) %>% 
  fromList() %>% 
  UpSetR::upset(
    nsets = length(.), 
    order.by = "freq"
  )

GO.terms.pergeneset <- GO %>% 
  lapply(function(x) {
    x$Description
  }) 
```

## KEGG pathways

```{r}
runKEGG = function(geneset) {

  uv =   bitr(nodes, 
                fromType = "SYMBOL", toType = "ENTREZID", 
                OrgDb ="org.Hs.eg.db") %>% 
    .$ENTREZID
  
  bitr(geneset, 
       fromType = "SYMBOL", toType = "ENTREZID", 
     OrgDb ="org.Hs.eg.db") %>% 
  .$ENTREZID %>%
  enrichKEGG(
    organism = 'hsa',
    universe = uv,
    pvalueCutoff = 0.05) %>% 
  as_tibble 
  
}
```

```{r}
kegg <- genesets %>% 
  sapply(function(x) {runKEGG(x)}, simplify = F) 

kegg %>% 
  lapply(function(x) {x$Description}) %>% 
  fromList() %>% 
  UpSetR::upset()
```


## reactome pathways 
```{r}
runReactome = function(geneset, genesetname) {
  
  enrichPathway(
    gene = bitr(geneset, 
                fromType = "SYMBOL", toType = "ENTREZID", 
                OrgDb ="org.Hs.eg.db") %>% 
      .$ENTREZID,
    universe = bitr(nodes,
                    fromType = "SYMBOL", toType = "ENTREZID",
                    OrgDb= "org.Hs.eg.db") %>% 
      .$ENTREZID, # all genes in pcnet
    readable = TRUE, 
    pvalueCutoff = 0.05
  ) %>% 
  as_tibble 
}

# perform hypergeometic test for significant overrep of Reactome db
reactome <- 
  genesets %>% 
  sapply(runReactome, "reactome", simplify = F)

reactome %>% 
  lapply(function(x) {x$Description}) %>% 
  fromList() %>% 
  UpSetR::upset(
    order.by = "freq", 
    nsets = length(.)
  )
```
## plot all results
```{r}
# a function to use complex heatmap version of upset to plot overlap of pathways
plotUpset = function(geneset_object, plot.title) 
{
  geneset_object %>% 
    lapply(function(x) {x$Description}) %>%
    enframe(name = "geneset", value = "gene") %>%
    unnest(gene) %>% 
    mutate(present = 1) %>% 
    pivot_wider(
      names_from = geneset,
      values_from = present,
      values_fill = 0
    ) %>% 
    unnest %>% 
    ComplexUpset::upset(
      intersect = names(geneset_object), 
      wrap = TRUE, 
      name = "Number of shared pathways"
    ) + 
    ggtitle(plot.title)
}
```


```{r}
plotUpset(geneset_object = reactome, plot.title = "Reactome genesets")
plotUpset(geneset_object = GO, plot.title = "Gene ontologies")
plotUpset(geneset_object = kegg, plot.title = "KEGG pathways") 
```


# Is there significant colocalisation?

Netcoloc was performed to determine whether there was significant network colocaliasaion between the CDs and adult dementias or brain malformations

As a positive control, we also ran AD vs PD, where known changes in the endolysosmal system & inflammation occurs. 

```{r}
# Read enrichment files
enrichment_dfs <- 
  list.files(
    "data/results/adultchild", recursive = TRUE, full.names = TRUE) %>% 
  grep(pattern = "network_colocalization_enrichment", 
       value = TRUE) %>%
  set_names(
    str_replace(., 
               pattern = "/network_colocalization_enrichment.csv",
               replacement = ""
  ) %>% 
    str_remove(
      pattern = paste0("data/results/adultchild", "/")
    )
  ) %>%   
  map(read.csv) %>%
  imap(~ mutate(.x, comparison = .y)) %>% 
   bind_rows() %>%
   dplyr::filter(z_comb == 3, z_12 == 1.5) %>%
   pivot_longer(
     cols = c(observed_overlap, expected_overlap_mean),
     names_to = "type",
     values_to = "value"
   ) %>%
   mutate(
     type = recode(type,
                   "observed_overlap" = "observed",
                   "expected_overlap_mean" = "expected")
     )

enrichment_dfs %>% 
  mutate(
    comparison = str_replace(
      comparison, 
      pattern = "CD_all_adult_all", 
      replacement = "childhood dementia\nvs\n adult-onset NDD"
    ), 
    comparison = str_replace(
      comparison, 
      pattern = "CD_all_brain_malform", 
      replacement = "childhood dementia\nvs\nbrain malformations"
    ),
    comparison = str_replace(
      comparison, 
      pattern = "AD_PD", 
      replacement = "Alzheimer's disease\nvs\nParkinson's disease"
    ),
    
  ) %>% 
  ggplot(
    aes(x = type, y = value, fill = type)
    ) +
  geom_bar(stat = "identity", position = "dodge", width = 0.6) +
  geom_errorbar(
    data = . %>% 
      filter(type == "expected"),
    aes(ymin = value - 2 * expected_overlap_std,
        ymax = value + 2 * expected_overlap_std),
    width = 0.25
  ) +
  geom_label(
    data =  . %>% 
      filter(type == "expected"),
    aes(x = 1.5, y = 440, label = sprintf("p = %.2e", empirical_p)),
    inherit.aes = FALSE,
    vjust = 0,
    hjust = 0.5,
    fontface = "bold",
    show.legend = FALSE
  ) +
  scale_y_continuous(
    limits = c(0,450), 
    breaks = seq(0,550, by = 100)
    ) +
  labs(
    y = "Number of genes", 
    x = ""
  ) +
  facet_wrap(~comparison, scales = "free_x") +
  theme(legend.position = "none") +
  scale_fill_manual(values = c("grey50", "skyblue"))
```

# Is the subnetwork biologically meaningful?
## Mouse variant phenotypes

I tested the enrichment of genes associated with mouse brain/neural phenotypes in the NetColoc subnetwork as a *whole*. I also included here a negative control phenotype -  abnormal digit morphology. This is not expected to be associated with the subnetwork. 

```{r, fig.width=12, fig.height=5, fig.fullwidth=TRUE}
list.files(
    "data/results/adultchild", recursive = TRUE, full.names = TRUE) %>% 
  grep(pattern = "/mousevariant_KO_enrichment.csv", 
       value = TRUE) %>% 
  set_names(
    str_replace(., 
               pattern = "/mousevariant_KO_enrichment.csv",
               replacement = ""
  ) %>% 
    str_remove(
      pattern = paste0("data/results/adultchild", "/")
    )
  ) %>%   
  map(read.csv) %>%
  imap(~ mutate(.x, comparison = .y)) %>% 
  bind_rows() %>% 
 dplyr::filter(comparison != "CD_all_brain_malform") %>%
  as_tibble() %>%
  group_by(comparison) %>%
  mutate(
    FDR = p.adjust(OR_p),
    .after = OR_p
  ) %>%
  group_by(comparison, MPO_term_type) %>%
  slice_min(order_by = OR_p, n = 10) %>%
  ungroup() %>%
  mutate(
    MP_description = reorder_within(MP_description, log_OR, comparison)
  ) %>%
  ggplot(
    aes(x = log_OR, y = MP_description)
  ) +
  geom_errorbarh(
    aes(xmin = log_OR_CI_lower, xmax = log_OR_CI_upper),
    height = 0.2, color = "black"
  ) +
  geom_point(
    aes(size = num_genes_in_term, color = MPO_term_type),
    alpha = 0.6
  ) +
  geom_vline(xintercept = 0, lty = 2, color = "grey50") +
  geom_label(
    aes(label = signif(OR_p, 2)),
    nudge_y = 0.4
  ) +
  scale_y_reordered() +  # This removes the `___comparison` suffix added by reorder_within()
  labs(
    x = "log(OR) Â± 95% CI",
    y = NULL,
    size = "# Genes in MPO term",
    color = "Term Type"
  ) +
  facet_wrap(~comparison, scales = "free")
```

## are the genes in the subnetwork human brain expressed?

### obtain the genes in the subnetworks
```{r}
node_metadata <- 
  list.files(path = "data/results/adultchild", 
             pattern = "overlapping_nodes_metadata", recursive = TRUE, full.names = TRUE) %>% 
  set_names(
    str_replace(., 
                pattern = "/overlapping_nodes_metadata.csv",
                replacement = ""
    ) %>% 
      str_remove(
        pattern = paste0("data/results/adultchild", "/")
      )
  ) %>%   
  map(read.csv) %>%
  imap(~ mutate(.x, comparison = .y))
```

```{r}
# plot number of genes per subnetworks
node_metadata %>% 
  bind_rows() %>% 
  dplyr::filter(comparison != "CD_all_brain_malform") %>% 
  dplyr::rename(gene = X) %>% 
  as_tibble() %>%
  mutate(
    seed_type = case_when(
      d1_seeds == 1 & d2_seeds == 1 ~ "both",
      d1_seeds == 1 & d2_seeds == 0 ~ "D1",
      d1_seeds == 0 & d2_seeds == 1 ~ "D2",
      d1_seeds == 0 & d2_seeds == 0 ~ "neither"
    )
  ) %>% 
  ggplot(
    aes(y = comparison, fill = seed_type)
  ) +
  geom_bar() +
  scale_fill_manual(values = c(
    "#7570b3", "#1b9e77", "#d95f02", "grey70"
  )) +
  labs(
    x = "Number of genes in subnetwork", 
    y = "Seed gene sets", 
    title = "Number of genes in the sub networks"
  ) 
```

### obtain GTEx expression data
This file was downloaded from here :
https://www.gtexportal.org/home/downloads/adult-gtex/bulk_tissue_expression

The file contains the median gene-level TPM by tissue. 

```{r}
# this was downloaded from gtex websiote
# version 10 
gtex <- read.delim("data/gtex/GTEx_Analysis_v10_RNASeQCv2.4.2_gene_median_tpm.gct.gz", skip =2)

gtex %<>% 
  pivot_longer(names_to = "tissue", 
               cols = 3:56, 
               values_to = "tpm"
  ) %>% 
  dplyr::select(
    gene = Description, tissue, tpm
  ) %>% 
  mutate(
    logtpm = log(tpm))
```

```{r}
# plot expression across tissues in gtex
# gtex %>% 
#   ggplot(
#     aes(x = logtpm)
#   ) +
#   geom_density() +
#   facet_wrap(~tissue, scales = "free") 
#   coord_cartesian(xlim = c(0,100))
  
# define a threshold for expressed or not  
gtex.threshold = 0.5


# Plot: proportion of genes expressed per brain tissue
bind_rows(
  gtex %>%
    mutate(expressed = ifelse(logtpm > gtex.threshold, 1, 0)) %>%
    filter(
      gene %in% node_metadata$CD_all_adult_all$X,
      grepl("Brain", tissue)
    ) %>%
    group_by(tissue) %>%
    summarise(
      n_genes = n(),
      n_expressed = sum(expressed),
      proportion_expressed = n_expressed / n_genes
    ) %>%
    mutate(subnetwork = "CD + Adult"), 
  gtex %>%
    mutate(expressed = ifelse(logtpm > gtex.threshold, 1, 0)) %>%
    filter(
      gene %in% node_metadata$AD_PD$X,
      grepl("Brain", tissue)
    ) %>%
    group_by(tissue) %>%
    summarise(
      n_genes = n(),
      n_expressed = sum(expressed),
      proportion_expressed = n_expressed / n_genes
    ) %>%
  mutate(subnetwork = "AD + PD")
) %>% 
ggplot(
  aes(
    x = proportion_expressed,
    y = reorder(tissue, proportion_expressed),
    fill = subnetwork
  )) +
  geom_col(position = position_dodge(width = 0.8), width = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, 0.2), limits = c(0, 1)) +
  labs(
    x = "Proportion of Genes Expressed",
    y = "Brain Tissue",
    fill = "Subnetwork",
    title = "Expression of Subnetwork Genes in Brain Tissue (GTEx)"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    axis.text.y = element_text(size = 10),
    panel.grid.major.y = element_blank(),
    plot.title = element_text(face = "bold")
  )
```

Approx 80% of the genes in the subnetworks are expressed in brain for both comparitor diseases. This is good IMO. 

# what pathways are in the subnetwork

```{r}
cluster.gprofiler.results <- 
  list.files(path = "data/results/adultchild", 
             pattern = "gprofiler", recursive = TRUE, full.names = TRUE) %>% 
  set_names(
    str_replace(., 
                pattern = "/gprofiler_analysis_results.csv",
                replacement = ""
    ) %>% 
      str_remove(
        pattern = paste0("data/results/adultchild", "/")
      )
  ) %>%   
  map(read.csv) %>%
  imap(~ mutate(.x, comparison = .y))
```

```{r}
cluster.gprofiler.results[c("AD_PD", "CD_all_adult_all")] %>% 
  lapply(function(x) {
    x %>% 
      mutate(
        FDR = p.adjust(p_value, method = "fdr")
      ) %>% 
      dplyr::filter(p_value != "NA") %>% 
      dplyr::distinct(Top.pathway, .keep_all = TRUE) %>% 
      head(10) %>% 
      mutate(
        Top.pathway = str_wrap(Top.pathway, width = 30)
      ) %>%
      ggplot(
        aes(
          x = -log10(FDR),
          y = reorder(Top.pathway, -p_value),
          fill = )
      ) +
      geom_col() +
      facet_wrap(~comparison)
  }) %>% 
  ggpubr::ggarrange(
    plotlist = .) 
```

# what genes are in the subnetworks which are drug targetable?

Here, i obtained the drug targets data from [omicsynth](https://doi.org/10.1016/j.ajhg.2023.12.006) to see what genes in the subnetwork have a known drug target. 

The plots below show the network proximity score (higher means more proximal to network) vs the evidence of drug interaction (higher interaction = higher score)

```{r}
# the genes in the subnet
ADCD.geneprox <- node_metadata$AD_PD %>% 
  dplyr::select(gene_name = X, z_both) %>% 
  as_tibble

CD_adult.geneprox <- node_metadata$CD_all_adult_all %>% 
  dplyr::select(gene_name = X, z_both) %>% 
  as_tibble

omicsynth <- read.csv("data/omicsynth/drug_genome_dgidb.csv") 

omicsynth %>% 
  dplyr::filter(gene_name %in% ADCD.geneprox$gene_name) %>%
  as_tibble %>% 
  left_join(ADCD.geneprox) %>% 
  ggplot(
    aes(x = interaction_group_score, y = z_both)
  ) +
  geom_point() +
  geom_label_repel(
    aes(label = drug_claim_name), 
    data = . %>% 
      dplyr::filter(z_both > 5 & interaction_group_score > 5)
  ) +
  labs(
    x = "Interaction group score", 
    y = "Network proximity z score", 
    title = "AD and PD subnetwork"
  )

omicsynth %>% 
  dplyr::filter(gene_name %in% CD_adult.geneprox$gene_name) %>%
  as_tibble %>% 
  left_join(CD_adult.geneprox) %>% 
  ggplot(
    aes(x = interaction_group_score, y = z_both)
  ) +
  geom_point() +
  geom_label_repel(
    aes(label = drug_claim_name), 
    data = . %>% 
      dplyr::filter(z_both > 10 & interaction_group_score > 10)
  ) +
  labs(
    x = "Interaction group score", 
    y = "Network proximity z score", 
    title = "Adult and childhood dementia subnetwork"
  )
```









